tasks/libs/ciproviders/gitlab_api.py

```diff
@@ -116,20 +116,48 @@ def refresh_pipeline(pipeline: ProjectPipeline):
 
 
 class GitlabCIDiff:
-    def __init__(self, before: dict, after: dict) -> None:
+    def __init__(
+        self, before: dict | None = None, after: dict | None = None, dict_initialization: dict | None = None
+    ) -> None:
         """
         Used to display job diffs between two gitlab ci configurations
         """
-        self.before = before
-        self.after = after
-        self.added_contents = {}
-        self.modified_diffs = {}
+        if dict_initialization:
```

Comment:
💬 I am wondering if having `GitlabCIDiff` inherit from a [UserDict](https://docs.python.org/3/library/collections.html#collections.UserDict) could make it more natural